paparazzi/src/main/java/app/cash/paparazzi/accessibility/RenderSettings.kt

```diff
@@ -41,20 +38,24 @@ internal object RenderSettings {
   private val colorMap = mutableMapOf<Int, Color>()
 
   fun getColor(view: View): Color {
-    if (view.id == View.NO_ID) {
-      view.id = generateViewId()
-    }
-    return getColor(view.id)
+    val hashCode = "${view::class.simpleName}(${view.iterableTextForAccessibility})".hashCode()
+    return getColor(hashCode)
   }
 
-  private fun getColor(viewId: Int): Color {
-    return colorMap.getOrPut(viewId) {
-      nextColor(viewId).withAlpha(DEFAULT_RENDER_ALPHA)
+  private fun getColor(hashCode: Int): Color {
+    return colorMap.getOrPut(hashCode) {
+      nextColor(hashCode).withAlpha(DEFAULT_RENDER_ALPHA)
     }
   }
 
-  private fun nextColor(viewId: Int): Color {
-    return DEFAULT_RENDER_COLORS[viewId  % DEFAULT_RENDER_COLORS.size]
+  private fun nextColor(hashCode: Int): Color {
+    return DEFAULT_RENDER_COLORS[colorIndex(hashCode)]
+  }
+
+  private fun colorIndex(hashCode: Int): Int {
+    val size = DEFAULT_RENDER_COLORS.size
+    val i = hashCode % size
+    return if (i < 0) i + size else i
```

Comment:
i know this is general hashcode practice, but can the hash code of a String ever be negative in practice?

if not, let's inline back to how it was?

Reply:
```
jshell> "jrodbx".hashCode()
$1 ==> -1151589837
```